docs/resources/servergroup.md

```diff
@@ -12,29 +11,139 @@ Represent server group resource
+  name       = "my-servergroup"
+  policy     = "affinity"
+  project_id = data.gcore_project.project.id
+  region_id  = data.gcore_region.region.id
+}
+```
+
+### Creating server group with two instances
+
+```terraform
+data "gcore_image" "ubuntu" {
+  name       = "ubuntu-22.04-x64"
+  region_id  = data.gcore_region.region.id
+  project_id = data.gcore_project.project.id
+}
+
+resource "gcore_volume" "boot_volume" {
+  name       = "my-boot-volume"
+  type_name  = "ssd_hiiops"
+  size       = 5
+  image_id   = data.gcore_image.ubuntu.id
+  project_id = data.gcore_project.project.id
+  region_id  = data.gcore_region.region.id
+}
+
+resource "gcore_volume" "boot_volume2" {
+  name       = "my-boot-volume2"
+  type_name  = "ssd_hiiops"
+  size       = 5
+  image_id   = data.gcore_image.ubuntu.id
+  project_id = data.gcore_project.project.id
+  region_id  = data.gcore_region.region.id
+}
+
+resource "gcore_instancev2" "instance" {
+  flavor_id     = "g1-standard-2-4"
+  name          = "my-instance"
+
+  volume {
+    volume_id  = gcore_volume.boot_volume.id
+    boot_index = 0
+  }
+
+  interface {
+    type = "external"
+    name = "my-external-interface"
+  }
+
+  project_id = data.gcore_project.project.id
+  region_id  = data.gcore_region.region.id
+}
+
+resource "gcore_instancev2" "instance2" {
+  flavor_id     = "g1-standard-2-4"
+  name          = "my-instance2"
+
+  volume {
+    volume_id  = gcore_volume.boot_volume2.id
+    boot_index = 0
+  }
+
+  interface {
+    type = "external"
+    name = "my-external-interface"
+  }
+
+  project_id = data.gcore_project.project.id
+  region_id  = data.gcore_region.region.id
+}
+
+resource "gcore_servergroup" "servergroup" {
   name       = "default"
   policy     = "affinity"
-  region_id  = 1
-  project_id = 1
+  project_id = data.gcore_project.project.id
+  region_id  = data.gcore_region.region.id
+  instance {
+      instance_id = gcore_instancev2.instance.id
+  }
+
+  instance {
+      instance_id = gcore_instancev2.instance2.id
+  }
 }
 ```
 
+
 <!-- schema generated by tfplugindocs -->
 ## Schema
 
 ### Required
 
 - `name` (String) Displayed server group name
-- `policy` (String) Server group policy. Available value is 'affinity', 'anti-affinity'
+- `policy` (String) Server group policy. Available value is 'affinity' 'anti-affinity' 'soft-anti-affinity'. 
+
+**Affinity rule**:
+
+Strictly isolate instances to different physical servers, minimizing simultaneous failures for critical applications.
+Choose this for enhanced fault tolerance.
+
+**Soft Anti-affinity rule**:
+
+Attempts to restrict instances belonging to the server group to the same host. Where it is not possible
+to schedule all instances on one host, they will be scheduled together on as few hosts as possible.
+
+**Anti-affinity**: 
+
+Boost performance by grouping related instances together. They will be hosted on the same physical server for
+faster interaction, enhancing communication and resource sharing.
```

Comment:
Actually I didn't like soft anti affinity description, beacuse of "they will be scheduled together on as few hosts as possible."

so I decided to review using chat gpt, here is the result:

Affinity Rule:

Strictly isolates instances across different physical servers, minimizing simultaneous failures for critical applications. This rule is recommended for enhanced fault tolerance.

Soft Anti-affinity Rule:

Attempts to place instances belonging to the same server group on a single host. If this is not possible, instances will be scheduled on the minimum number of hosts required.

Anti-affinity Rule:

Improves performance by grouping related instances on the same physical server to enhance communication, resource sharing, and faster interaction.